crates/testing/src/spinning_task.rs

```diff
@@ -86,80 +79,120 @@ impl SpinningTaskDescription {
+                                                UpDown::NetworkDown => {
+                                                    if let Some(handle) = state.handles.get(idx) {
+                                                        tracing::error!(
+                                                            "Node {} networks pausing",
+                                                            idx
+                                                        );
+                                                        handle.networks.0.pause();
+                                                        handle.networks.1.pause();
+                                                    }
+                                                }
                                             }
                                         }
                                     }
+
+                                    // update our latest view
+                                    state.latest_view = Some(view_number);
                                 }
                             }
+
                             (None, state)
                         }
                         .boxed()
                     },
                 ));
+
+                let mut streams = vec![];
+                for handle in &mut state.handles {
+                    let s1 = handle
+                        .handle
+                        .get_event_stream_known_impl(FilterEvent::default())
+                        .await
+                        .0;
+                    let s2 = handle
+                        .handle
+                        .get_internal_event_stream_known_impl(FilterEvent::default())
+                        .await
+                        .0;
+                    streams.push(Merge::new(s1, s2));
+                }
                 let builder = TaskBuilder::<SpinningTaskTypes<TYPES, I>>::new(
                     "Test Spinning Task".to_string(),
                 )
                 .register_event_stream(test_event_stream, FilterEvent::default())
                 .await
                 .register_registry(&mut registry)
                 .await
-                .register_state(state)
-                .register_event_handler(event_handler)
                 .register_message_handler(message_handler)
-                .register_message_stream(stream_generator);
+                .register_message_stream(MergeN::new(streams))
+                .register_event_handler(event_handler)
+                .register_state(state);
                 let task_id = builder.get_task_id().unwrap();
                 (task_id, SpinningTaskTypes::build(builder).launch())
             }
             .boxed()
         })
     }
 }
+
+/// types for safety task
+pub type SpinningTaskTypes<TYPES, I> = HSTWithEventAndMessage<
+    SpinningTaskErr,
+    GlobalTestEvent,
+    ChannelStream<GlobalTestEvent>,
+    (usize, Either<Event<TYPES>, HotShotEvent<TYPES>>),
```

Comment:
Same as above, can't this just be `(usize,  HotShotEvent<TYPES>)`

Reply:
Fixed !